GitExtUtils/GitUI/ToolStripExtensions.cs

```diff
@@ -56,6 +56,6 @@ internal static void EnableTheming(this ToolStrip toolStrip, bool enable)
         }
 
         internal static bool IsThemingEnabled(this ToolStrip toolStrip) =>
-            toolStrip.Renderer is ToolStripExThemeAwareRenderer;
+            toolStrip.Renderer is ToolStripProfessionalRenderer;
```

Comment:
This seems to be working also in 3.5 where theming is possible.

Reply:
I'm not seeing `ToolStripExThemeAwareRenderer` used even in release/3.5 branch... So I think the types all references can be completely deleted.
The only question remains what was the intent for this type? It was introduced in #8899 but even there I can't see any instantiations. @NikolayXHD any thoughts?

Reply:
Hi @RussKie,

To give a little bit of context, it is actually created through reflection, via:
https://github.com/gitextensions/gitextensions/blob/d488e497cc6244483794abedb8e4779495ba861c/GitExtUtils/GitUI/ToolStripExtensions.cs#L43

This code is used when using either `ToolStripEx`, through `UseCustomRenderer`, or through a call to `EnableTheming`, done by the `ThemeFix` class.

The main purpose of these two type is to extend WinForms renderers to allow registering a `IMenuItemBackgroundFilter`.
This feature is used for a single purpose in master, through:
https://github.com/gitextensions/gitextensions/blob/5190ba5c1a5f35b321295c59fb04e2a8ae454c14/GitUI/UserControls/RevisionGrid/MenuUtil.cs#L28-L31

This usage is done in the two locations that were having the bug, due to the `ToolStripExtensions` not considering the fact that if GitExtensions theming is not activated, then the actual type is the WinForms one, as the `ThemeFix` will not call the `EnableTheming`:
https://github.com/gitextensions/gitextensions/blob/3d4ed902fcd13dba41d66dd1e6b918ba78eac3f5/GitExtUtils/GitUI/Theming/ThemeFix.cs#L21-L26

Reply:
What @Greybird says is correct.

In addition to that, in Windows10 `MenuItemBackgroundFilter` appears to be unnecessary, which renders `OnRenderMenuItemBackground` methods obsolete.

I am going to test, if this is also unnecessary in Windows 7. Bases on the outcome, either a bunch of code will be removed, or the documentation will reflect that the code is required to render "caption" menu items correctly in Windows 7